sqlglot/dialects/tsql.py

```diff
@@ -502,6 +508,110 @@ class Parser(parser.Parser):
+                            )
+                        ],
+                    )
+                ],
+            ),
+            "MIN_GRANT_PERCENT": lambda self: self._generic_parse_option(
+                "MIN_GRANT_PERCENT",
+                [
+                    lambda: self._match(TokenType.EQ)
+                    and self._generic_parse_option(
+                        "MIN_GRANT_PERCENT",
+                        [
+                            lambda: self._return_option_with_value(
+                                "MIN_GRANT_PERCENT", self._parse_number()
+                            )
+                        ],
+                    )
+                ],
+            ),
+            "MAXDOP": lambda self: self._generic_parse_option(
+                "MAXDOP", [lambda: self._return_option_with_value("MAXDOP", self._parse_number())]
+            ),
+            "MAXRECURSION": lambda self: self._generic_parse_option(
+                "MAXRECURSION",
+                [lambda: self._return_option_with_value("MAXRECURSION", self._parse_number())],
+            ),
+            "NO_PERFORMANCE_SPOOL": lambda self: self._generic_parse_option(
+                "NO_PERFORMANCE_SPOOL", []
+            ),
+            "OPTIMIZE": lambda self: self._generic_parse_option(
+                "OPTIMIZE",
+                [
+                    # There is still no support for the following syntax:
+                    # OPTIMIZE FOR ( @variable_name { UNKNOWN | = <literal_constant> } [ , ...n ] )
+                    lambda: self._match_text_seq("FOR")
+                    and self._generic_parse_option("OPTIMIZE FOR", ["UNKNOWN"])
+                ],
+            ),
+            "PARAMETERIZATION": lambda self: self._generic_parse_option(
+                "PARAMETERIZATION", ["SIMPLE", "FORCED"]
+            ),
+            "QUERYTRACEON": lambda self: self._generic_parse_option(
+                "QUERYTRACEON",
+                [lambda: self._return_option_with_value("QUERYTRACEON", self._parse_number())],
+            ),
+            "RECOMPILE": lambda self: self._generic_parse_option("RECOMPILE", []),
+            "ROBUST": lambda self: self._generic_parse_option("ROBUST", ["PLAN"]),
+            "USE": lambda self: self._generic_parse_option(
+                "USE",
+                [
+                    lambda: self._match_text_seq("PLAN")
+                    and self._return_option_with_value("USE PLAN", self._parse_primary())
+                ],
+            ),
+            # No support yet for TABLE HINT
+        }
+
+        def _parse_string(self) -> t.Optional[exp.Expression]:
+            if self._curr is not None and self._curr.token_type == TokenType.NATIONAL_STRING:
+                result = self.PRIMARY_PARSERS[TokenType.NATIONAL_STRING](self, self._curr)
+                self._match(TokenType.NATIONAL_STRING)
+                return result
+            return super()._parse_string()
```

Comment:
Probably missed this one? If we keep the `_parse_primary()` in `USE` it should not be needed anymore.